### PR TITLE
Update csi hostpath driver in e2e

### PR DESCRIPTION
**Pull request by msau42**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Update hostpath csi driver to latest images

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by msau42** [comment]
cc @jsafrane @mkimuram 
Looks like the csi hostpath driver is failing for multivolume block tests. I suspect it is a driver issue so far, but haven't looked deeply into it yet. From an initial glance [between 1.1.0-rc1 and 1.1.0](https://github.com/kubernetes-csi/csi-driver-host-path/compare/v1.1.0-rc1...v1.1.0), I wasn't able to find anything suspicious yet.

**Comment by msau42** [comment]
Tests pass using hostpath 1.1.0-rc1.  Moving back to 1.1.0 to further debug

**Comment by msau42** [comment]
Hmm the CSI job running on kind passed with multiVolume raw block tests using driver 1.1.0:
https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/kubernetes-csi_csi-driver-host-path/66/pull-kubernetes-csi-csi-driver-host-path-unit/1148778607959609344

**Comment by msau42** [comment]
I0710 19:03:10.467620       1 server.go:117] GRPC call: /csi.v1.Controller/CreateVolume
I0710 19:03:10.468081       1 server.go:118] GRPC request: {"capacity_range":{"required_bytes":5368709120},"name":"pvc-6cfa94d0-1b15-4e27-8474-9e69d587f9be","volume_capabilities":[{"AccessType":{"Block":{}},"access_mode":{"mode":1}}]}
I0710 19:03:10.476189       1 controllerserver.go:152] failed to create block device: fallocate: cannot open /csi-data-dir/5c19703f-a345-11e9-8f2a-42010a280004: No such file or directory

File: test/e2e/testing-manifests/storage-csi/hostpath/hostpath/csi-hostpathplugin.yaml [hunk 1]
```diff
@@ -58,6 +58,8 @@ spec:
             - mountPath: /var/lib/kubelet/plugins
               mountPropagation: Bidirectional
               name: plugins-dir
+            - mountPath: /csi-data-dir
+              name: csi-data-dir
```
Comment (msau42) [inline test/e2e/testing-manifests/storage-csi/hostpath/hostpath/csi-hostpathplugin.yaml]: Requiring this path to be created by Kubernetes is a breaking change. Do you think it's useful to still support the mode where data is not persisted?

Comment (jsafrane) [reply test/e2e/testing-manifests/storage-csi/hostpath/hostpath/csi-hostpathplugin.yaml]: I think it won't harm to persist the data, however, do we really need it? Do we have a test that restarts the driver itself so it looses data? I must miss something here.

Comment (jsafrane) [reply test/e2e/testing-manifests/storage-csi/hostpath/hostpath/csi-hostpathplugin.yaml]: I read the code first and then your comments about testing :-). IMO, the driver should create `/csi-data-dir` if it does not exist.

Comment (msau42) [reply test/e2e/testing-manifests/storage-csi/hostpath/hostpath/csi-hostpathplugin.yaml]: Fixing in https://github.com/kubernetes-csi/csi-driver-host-path/pull/73.

For e2e testing, the only potential issue is that the data will leak on the node afterwards?

**Comment by msau42** [comment]
This is ready to go

File: test/e2e/testing-manifests/storage-csi/hostpath/hostpath/csi-hostpathplugin.yaml (hunk 1 above)
Comment (msau42) [reply test/e2e/testing-manifests/storage-csi/hostpath/hostpath/csi-hostpathplugin.yaml]: Updated to latest tag for hostpath driver to switch back to non-persistent data on driver restarts.

**Review by pohly — COMMENTED** [review]
It looks like you are just bumping the versions locally in `test/e2e/testing-manifests`, right? That means they are no longer consistent with the deployment files in the csi-driver-host-path.

As a stop-gap solution this is okay, but I think long term we should get back to testing exactly what we publish in csi-driver-host-path. We can merge this here now and I can replace with verbatim copies as part of my pending PR https://github.com/kubernetes/kubernetes/pull/80568/.

**Comment by msau42** [comment]
Bumping just the versions caught a backwards incompatible change in the driver: kubernetes-csi/csi-driver-host-path#73

I also think that for our e2es we don't actually want the behavior of persisting the data to local disk.  I want to eventually add CSI compatibility to the k8s conformance tests and use hostpath CSI driver.

**Comment by pohly** [comment]
> Bumping just the versions caught a backwards incompatible change in the driver [...] I also think that for our e2es we don't actually want the behavior of persisting the data to local disk.

So it is intentionally that E2E testing isn't using a verbatim copy of the upstream deployment files? I'm still confused about what the goal is regarding these deployments.

**Comment by msau42** [comment]
It's not intentional but this is what my current thoughts are of the direction of these tests.

The main purpose of testing hostpath csi in k/k is so that we make sure that changes in kubernetes do not break csi.  The purpose of testing hostpath driver in kubernetes-csi is slightly different. There we want to test our changes to sidecars.

**Comment by pohly** [comment]
> It's not intentional but this is what my current thoughts are of the direction of these tests.
>
> The main purpose of testing hostpath csi in k/k is so that we make sure that changes in kubernetes do not break csi. 

Which means we still want to test the upstream deployment and will get there again eventually. Makes sense to me, so...

/lgtm

**Pull request merged**